### PR TITLE
Figure.coast: Add alias "box" for "-F"

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

This PR aims to add the alias `box` for **-F** of `pygmt.Figure.coast`:

- [x] Add to alias list.
- [x] **Do we want a consistent alphabetical order?**  -> see discussion in issue #884
- [x] Copy-past docstring from `pygmt.Figure.basemap`.
- [ ] Compare and adjust regarding differences.
      https://docs.generic-mapping-tools.org/dev/basemap.html#f vs. https://docs.generic-mapping-tools.org/dev/coast.html#f
- [x] **Do we want to add this docstring to `decorators.py`?** Have to check whether this makes sense. There are other methods that also have the `box` parameter, but there are probably differences in the functionality and syntax. -> No

**Preview**: https://pygmt-dev--2823.org.readthedocs.build/en/2823/api/generated/pygmt.Figure.coast.html

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by seisman** [comment]
> * **Do we want to add this docstring to `decorators.py`?** Have to check whether this makes sense. There are other methods that also have the `box` parameter, but there are probably differences in the functionality and syntax.
> 
> **Preview**:

I believe the `box` parameter has different syntax in different modules (like `basemap`, `image` and `coast`).

**Comment by seisman** [comment]
> * Do we want a consistent alphabetical order?

I think we're sorting the single-letter options alphabetically in most module wrappers.

**Comment by yvonnefroehlich** [comment]
> > * Do we want a consistent alphabetical order?
> 
> I think we're sorting the single-letter options alphabetically in most module wrappers.

For `coast`, this is done in commit https://github.com/GenericMappingTools/pygmt/pull/2823/commits/a32e08a59a0b8668918d5b2b83316425b78e2bdd. Should I submit a separate PR to make this consistent accross als functions and methods?

**Comment by seisman** [comment]
> > > * Do we want a consistent alphabetical order?
> > 
> > 
> > I think we're sorting the single-letter options alphabetically in most module wrappers.
> 
> For `coast`, this is done in commit [a32e08a](https://github.com/GenericMappingTools/pygmt/commit/a32e08a59a0b8668918d5b2b83316425b78e2bdd). Should I submit a separate PR to make this consistent accross als functions and methods?

Sorry, it seems that we still haven't reached an agreement about the ordering of aliases. See https://github.com/GenericMappingTools/pygmt/issues/884.

**Comment by yvonnefroehlich** [comment]
> > > > * Do we want a consistent alphabetical order?
> > > 
> > > 
> > > I think we're sorting the single-letter options alphabetically in most module wrappers.
> > 
> > 
> > For `coast`, this is done in commit [a32e08a](https://github.com/GenericMappingTools/pygmt/commit/a32e08a59a0b8668918d5b2b83316425b78e2bdd). Should I submit a separate PR to make this consistent accross als functions and methods?
> 
> Sorry, it seems that we still haven't reached an agreement about the ordering of aliases. See #884.

No worries, thanks for pointing out this issue! I will have a more detailed look at it in the next days.

**Comment by yvonnefroehlich** [comment]
> @yvonnefroehlich What's the status of this PR?

So far I simply copied the docstring for the `box` parameter of `Figure.basemap` and did not compare https://docs.generic-mapping-tools.org/dev/basemap.html#f vs. https://docs.generic-mapping-tools.org/dev/coast.html#f. However, if people know that the provided functionaly is identical, this PR should be ready for review.

**Review by seisman — APPROVED** [review]

File: pygmt/src/coast.py [hunk 1]
```diff
@@ -128,6 +129,24 @@ def coast(self, **kwargs):
...
+    box : bool or str
+        [**+c**\ *clearances*][**+g**\ *fill*][**+i**\ [[*gap*/]\ *pen*]]\
+        [**+p**\ [*pen*]][**+r**\ [*radius*]][**+s**\ [[*dx*/*dy*/][*shade*]]].
+        If set to ``True``, draw a rectangular border around the
+        map scale or rose. Alternatively, specify a different pen with
+        **+p**\ *pen*. Add **+g**\ *fill* to fill the scale panel [Default is
+        no fill]. Append **+c**\ *clearance* where *clearance* is either gap,
+        xgap/ygap, or lgap/rgap/bgap/tgap where these items are uniform,
+        separate in x- and y-direction, or individual side spacings between
+        scale and border. Append **+i** to draw a secondary, inner border as
```
[the remark this answers is not in the document]
Comment (seisman) [reply pygmt/src/coast.py:141]: Using plural makes more sense to me.

[the remark this answers is not in the document]
Comment (yvonnefroehlich) [reply pygmt/src/coast.py]: Updated to "clearances" (plural) for `Figure.basemap` and `Figure.coast` in commit https://github.com/GenericMappingTools/pygmt/pull/2823/commits/9be73a9b592a1031879e06a329114db138370e9f.

But there are actually more affected docstrings. Maybe we should move this change to a separate PR and focus in this PR on adding the alias `box` for **-F**?
```
(base) yfroe@gpiseis16:~/Documents/01_GitHub/github_pygmt/pygmt/pygmt/src> grep "clearance" *
basemap.py:        [**+c**\ *clearances*][**+g**\ *fill*][**+i**\ [[*gap*/]\ *pen*]]\
basemap.py:        no fill]. Append **+c**\ *clearance* where *clearance* is either gap,
colorbar.py:        [**+c**\ *clearances*][**+g**\ *fill*][**+i**\ [[*gap*/]\ *pen*]]\
colorbar.py:        **+c**\ *clearance* where *clearance* is either gap, xgap/ygap, or
image.py:        [**+c**\ *clearances*][**+g**\ *fill*][**+i**\ [[*gap*/]\ *pen*]]\
inset.py:        [**+c**\ *clearances*][**+g**\ *fill*][**+i**\ [[*gap*/]\
inset.py:        Append **+c**\ *clearance* where *clearance* is either
inset.py:        This is clearance that is added around the inside of the inset.
legend.py:        [**+c**\ *clearances*][**+g**\ *fill*][**+i**\ [[*gap*/]\ *pen*]]\
grep: __pycache__: Is a directory
subplot.py:    C="clearance",
subplot.py:        **+c**\ *dx*\[/*dy*] to set the clearance between the tag and a
subplot.py:    clearance : str or list
subplot.py:        [*side*]\ *clearance*.
subplot.py:        Reserve a space of dimension *clearance* between the margin and the
subplot.py:        **s** and **n**. No *side* means all sides (i.e. ``clearance="1c"``
subplot.py:        would set a clearance of 1 cm on all sides). The option is repeatable
subplot.py:        ``clearance=["w1c", "s2c"]`` would set a clearance of 1 cm on west
subplot.py:        ``clearance`` to secure extra space for long horizontal annotations.
subplot.py:@use_alias(A="fixedlabel", C="clearance", V="verbose")
subplot.py:    clearance : str or list
subplot.py:        [*side*]\ *clearance*.
subplot.py:        Reserve a space of dimension *clearance* between the margin and the
subplot.py:        than one side (e.g. ``clearance=["w1c", "s2c"]`` would set a clearance
subplot.py:        clearances set by ``clearance`` in the initial
text.py:    C="clearance",
text.py:    clearance : str
text.py:        Adjust the clearance between the text and the surrounding box
text.py:        (see ``clearance``) [Default is ``"0.25p,black,solid"``].
```

[the remark this answers is not in the document]
Comment (yvonnefroehlich) [reply pygmt/src/coast.py:141]: OK. Then we leave this as is and this PR should be finished. I revert commit https://github.com/GenericMappingTools/pygmt/pull/2823/commits/9be73a9b592a1031879e06a329114db138370e9f.

**Pull request merged**